roachtest: drop/tpcc/w=100,nodes=9 failed on master

SHA: https://github.com/cockroachdb/cockroach/commits/83e98d996b3ba3537afd7bc67e439eb5d3ea1659

Parameters:

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=683712&tab=buildLog

```
	drop.go:139,cluster.go:960,errgroup.go:58: Node 3 space after deletion used: 5.7 GiB
		disk space usage has not dropped below 100 MiB on all nodes.

```

Most recent failure is:

```
panic: runtime error: index out of range [recovered]
	panic: panic while executing 1 statements: DROP DATABASE _; caused by runtime error: index out of range

goroutine 24013 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc42beca600, 0x2e3de80, 0xc42a0c9740, 0x25cf040, 0x3f2de20)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:674 +0x36f
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc42beca600, 0x2e3de80, 0xc42a0c9740)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:414 +0x61
panic(0x25cf040, 0x3f2de20)
	/usr/local/go/src/runtime/panic.go:502 +0x229
github.com/cockroachdb/cockroach/vendor/golang.org/x/net/trace.(*trace).addEvent(0xc429547ba0, 0x2590860, 0xc42c64a5d0, 0xc42ab90000)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/net/trace/trace.go:772 +0x734
github.com/cockroachdb/cockroach/vendor/golang.org/x/net/trace.(*trace).LazyPrintf(0xc429547ba0, 0x290309e, 0x2, 0xc42ab938a0, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/net/trace/trace.go:801 +0xa0
github.com/cockroachdb/cockroach/pkg/util/tracing.(*span).LogFields(0xc44f43c1c0, 0xc435b89d40, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/tracer_span.go:390 +0x658
github.com/cockroachdb/cockroach/pkg/util/log.eventInternal(0x2e3df40, 0xc44489aed0, 0x100, 0x292dcb0, 0x18, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/trace.go:131 +0x277
github.com/cockroachdb/cockroach/pkg/util/log.vEventf(0x2e3df40, 0xc44489aed0, 0xc43577d100, 0x1, 0x2, 0x292dcb0, 0x18, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/trace.go:178 +0x13c
github.com/cockroachdb/cockroach/pkg/util/log.VEventf(0x2e3df40, 0xc44489aed0, 0xc400000002, 0x292dcb0, 0x18, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/trace.go:193 +0x8d
github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).rollback(0xc44f5f59a0, 0x2e3df40, 0xc44489aed0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:584 +0xa0
github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).CleanupOnError(0xc44f5f59a0, 0x2e3df40, 0xc44489aed0, 0x2e119a0, 0xc4200801e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:510 +0x6b
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn(0xc42047f700, 0x2e3df40, 0xc44489aed0, 0xc42c64a5a0, 0xc42beca601, 0xc42c64a5a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:583 +0x197
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn-fm(0x2e3df40, 0xc44489aed0, 0xc42c64a5a0, 0xc43577d400, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/physical_schema_accessors.go:187 +0x48
github.com/cockroachdb/cockroach/pkg/sql.(*databaseCache).getDatabaseDesc(0xc42d6ada40, 0x2e3df40, 0xc44489aed0, 0xc43577d4e0, 0xc44f436258, 0x4, 0x6c1900, 0xc42c64a570, 0x30, 0x28)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/database.go:208 +0x199
github.com/cockroachdb/cockroach/pkg/sql.(*databaseCache).getDatabaseID(0xc42d6ada40, 0x2e3df40, 0xc44489aed0, 0xc43577d4e0, 0xc44f436258, 0x4, 0x7fd6b4308400, 0xc43577d4f0, 0x6db169, 0x8)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/database.go:246 +0xaa
github.com/cockroachdb/cockroach/pkg/sql.(*TableCollection).waitForCacheToDropDatabases.func1(0xc42d6ada40, 0x2a55268)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:403 +0x92
github.com/cockroachdb/cockroach/pkg/sql.(*databaseCacheHolder).waitForCacheState(0xc420389160, 0xc42c64a570)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/exec_util.go:411 +0x6e
github.com/cockroachdb/cockroach/pkg/sql.(*TableCollection).waitForCacheToDropDatabases(0xc42beca838, 0x2e3df40, 0xc44489aed0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:400 +0x111
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).txnStateTransitionsApplyWrapper(0xc42beca600, 0x2e16960, 0x4535148, 0x267eee0, 0xc43d205150, 0x7fd6b2e51450, 0xc44f4402d0, 0x260, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1866 +0x478
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc42beca600, 0x2e3de80, 0xc42a0c9740, 0xc42cdfe300, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1203 +0xda3
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc42004f9e0, 0x2e3de80, 0xc42a0c9740, 0x0, 0x0, 0xc42db4e01e, 0x4, 0x0, 0x0, 0x2e1ffc0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:416 +0x1bb
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl.func3(0xc42004f9e0, 0x2e3de80, 0xc42a0c9740, 0xc42b6a01c0, 0x5400, 0x15000, 0xc42052f140, 0xc42cdfe300, 0xc42cdfe2f0, 0xc4327a1bb0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:267 +0x122
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:266 +0xf04
```

This is something in sql land, perhaps related to the schema change machinery. @vivekmenezes Can you triage?

last crash happened at the same spot

```
E180822 07:22:57.074567 12143 sql/conn_executor.go:661  [n1,client=35.196.161.33:46682,user=root] a SQL panic has occurred while executing "DROP DATABASE tpcc": runtime error: index out of range
E180822 07:22:57.074693 12143 util/log/crash_reporting.go:203  [n1,client=35.196.161.33:46682,user=root] a panic has occurred!
E180822 07:22:57.301910 12143 util/log/crash_reporting.go:477  [n1,client=35.196.161.33:46682,user=root] Reported as error 6a57aacb46c14bfd965074783c5f822f
panic: runtime error: index out of range [recovered]
	panic: panic while executing 1 statements: DROP DATABASE _; caused by runtime error: index out of range

goroutine 12143 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc429c5c000, 0x2f048c0, 0xc42d578fc0, 0x2670c80, 0x4095dc0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:675 +0x36f
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc429c5c000, 0x2f048c0, 0xc42d578fc0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:414 +0x61
panic(0x2670c80, 0x4095dc0)
	/usr/local/go/src/runtime/panic.go:502 +0x229
github.com/cockroachdb/cockroach/vendor/golang.org/x/net/trace.(*trace).addEvent(0xc428f37a00, 0x2631e60, 0xc42bdb1c80, 0xc435980000)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/net/trace/trace.go:772 +0x734
github.com/cockroachdb/cockroach/vendor/golang.org/x/net/trace.(*trace).LazyPrintf(0xc428f37a00, 0x29befbf, 0x2, 0xc435980120, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/net/trace/trace.go:801 +0xa0
github.com/cockroachdb/cockroach/pkg/util/tracing.(*span).LogFields(0xc4380ee0e0, 0xc42ca77800, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/tracer_span.go:390 +0x658
github.com/cockroachdb/cockroach/pkg/util/log.eventInternal(0x2f04980, 0xc430ce4600, 0x100, 0x29ea0c8, 0x18, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/trace.go:131 +0x277
github.com/cockroachdb/cockroach/pkg/util/log.vEventf(0x2f04980, 0xc430ce4600, 0xc42b01f100, 0x1, 0x2, 0x29ea0c8, 0x18, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/trace.go:178 +0x13c
github.com/cockroachdb/cockroach/pkg/util/log.VEventf(0x2f04980, 0xc430ce4600, 0xc400000002, 0x29ea0c8, 0x18, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/trace.go:193 +0x8d
github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).rollback(0xc42a4f8500, 0x2f04980, 0xc430ce4600, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:598 +0xa0
github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).CleanupOnError(0xc42a4f8500, 0x2f04980, 0xc430ce4600, 0x2ed7280, 0xc4200821e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:524 +0x6b
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn(0xc420712b80, 0x2f04980, 0xc430ce4600, 0xc42bdb1c50, 0xc429c5c001, 0xc42bdb1c50)
	/go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:588 +0x197
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn-fm(0x2f04980, 0xc430ce4600, 0xc42bdb1c50, 0xc42b01f400, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/physical_schema_accessors.go:187 +0x48
github.com/cockroachdb/cockroach/pkg/sql.(*databaseCache).getDatabaseDesc(0xc42e3862c0, 0x2f04980, 0xc430ce4600, 0xc42b01f4e0, 0xc470c79d20, 0x4, 0x6c1900, 0xc42bdb1c20, 0x30, 0x28)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/database.go:208 +0x199
github.com/cockroachdb/cockroach/pkg/sql.(*databaseCache).getDatabaseID(0xc42e3862c0, 0x2f04980, 0xc430ce4600, 0xc42b01f4e0, 0xc470c79d20, 0x4, 0x7fe507bbd000, 0xc42b01f4f0, 0x6db169, 0x8)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/database.go:246 +0xaa
github.com/cockroachdb/cockroach/pkg/sql.(*TableCollection).waitForCacheToDropDatabases.func1(0xc42e3862c0, 0x2b13b18)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:403 +0x92
github.com/cockroachdb/cockroach/pkg/sql.(*databaseCacheHolder).waitForCacheState(0xc420378560, 0xc42bdb1c20)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/exec_util.go:424 +0x6e
github.com/cockroachdb/cockroach/pkg/sql.(*TableCollection).waitForCacheToDropDatabases(0xc429c5c238, 0x2f04980, 0xc430ce4600)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:400 +0x111
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).txnStateTransitionsApplyWrapper(0xc429c5c000, 0x2edc880, 0x46a5530, 0x2725ca0, 0xc430f6a4b8, 0x7fe507b9a610, 0xc47057b440, 0x260, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1875 +0x478
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc429c5c000, 0x2f048c0, 0xc42d578fc0, 0xc4314bcee0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1204 +0xda3
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc4201ef0e0, 0x2f048c0, 0xc42d578fc0, 0x0, 0x0, 0xc42ed0101e, 0x4, 0x0, 0x0, 0x2ee6480, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:416 +0x1bb
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl.func3(0xc4201ef0e0, 0x2f048c0, 0xc42d578fc0, 0xc42944e000, 0x5400, 0x15000, 0xc42012b800, 0xc4314bcee0, 0xc4314bced0, 0xc44e65b590, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:267 +0x122
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:266 +0xf04
```

--- a/pkg/sql/conn_executor.go
+++ b/pkg/sql/conn_executor.go
@@ -1834,8 +1834,7 @@ func (ex *connExecutor) txnStateTransitionsApplyWrapper(
 		implicitTxn = os.ImplicitTxn.Get()
 	}
 
-	ctx := withStatement(ex.Ctx(), ex.curStmt)
-	err := ex.machine.ApplyWithPayload(ctx, ev, payload)
+	err := ex.machine.ApplyWithPayload(withStatement(ex.Ctx(), ex.curStmt), ev, payload)
 	if err != nil {
 		if _, ok := err.(fsm.TransitionNotFoundError); ok {
 			panic(err)
@@ -1872,7 +1871,7 @@ func (ex *connExecutor) txnStateTransitionsApplyWrapper(
 		}
 
 		// Wait for the cache to reflect the dropped databases if any.
-		ex.extraTxnState.tables.waitForCacheToDropDatabases(ctx)
+		ex.extraTxnState.tables.waitForCacheToDropDatabases(ex.Ctx())
 
 		fallthrough
 	case txnRestart, txnAborted:

